### ISSUE TITLE
Fable testing: Checkbox

### ISSUE THREAD

**Issue by connorscarolyns** [cds-snc/platform-forms-client#4850]
The checkbox component reads the label between each selection and announces content unexpectedly:

https://app.makeitfable.com/share/675aea6a4c64daabd8c056c9
https://app.makeitfable.com/share/675adf72b10d6161df915d94
https://app.makeitfable.com/share/6759c150a31758a9afc7a483
https://app.makeitfable.com/share/675ae5114c64daabd8c056c8

VoiceOver in Safari
VoiceOver in Chrome
NVDA in Chrome
NVDA in Edge

**Comment by thiessenp-cds** [comment]
@connorscarolyns if you have any free time, could you try to reproduce this issue using chrome and VoiceOver? Here is a form on staging you could use for testing:
https://forms-staging.cdssandbox.xyz/en/id/cm5ciqlzu0002xe69o0itybq7

I tried with Windows + Edge + NVDA and MacOS + VoiceOver. I can't get the checkbox description to read out when a check is updated. I can definitely see it from the user testing though. I may be missing something. 

https://github.com/user-attachments/assets/bbf31683-5c74-4e9c-9adc-8754442d2a3e

Actually would you have a link to the form they used for testing, that may help?

**Comment by thiessenp-cds** [comment]
Here is the test file from @connorscarolyns 
[lemonade-stand-funding (fable).json](https://github.com/user-attachments/files/18282439/lemonade-stand-funding.fable.json)

**Comment by thiessenp-cds** [comment]
I could reproduce the error. It looks like the error is the form-builder is copying the checkbox's legend element into each checkbox. 
![Screenshot 2024-12-31 at 10 33 26 AM](https://github.com/user-attachments/assets/8c822834-bdc9-402d-8eca-5611c843fc35)
![Screenshot 2024-12-31 at 10 35 47 AM](https://github.com/user-attachments/assets/e394d91e-f4b9-4cbe-bea4-00bbdbce6e8a)
